common/persistence/nosql/nosqlplugin/cassandra/gocql/consistency.go

```diff
@@ -80,3 +81,104 @@ func mustConvertSerialConsistency(c SerialConsistency) gocql.SerialConsistency {
 		panic(fmt.Sprintf("Unknown gocql SerialConsistency level: %v", c))
 	}
 }
+
+func (c Consistency) MarshalText() (text []byte, err error) {
+	return []byte(c.String()), nil
+}
+
+func (c *Consistency) UnmarshalText(text []byte) error {
+	switch string(text) {
+	case "ANY":
+		*c = Any
+	case "ONE":
+		*c = One
+	case "TWO":
+		*c = Two
+	case "THREE":
+		*c = Three
+	case "QUORUM":
+		*c = Quorum
+	case "ALL":
+		*c = All
+	case "LOCAL_QUORUM":
+		*c = LocalQuorum
+	case "EACH_QUORUM":
+		*c = EachQuorum
+	case "LOCAL_ONE":
+		*c = LocalOne
+	default:
+		return fmt.Errorf("invalid consistency %q", string(text))
+	}
+
+	return nil
+}
+
+func (c Consistency) String() string {
+	switch c {
+	case Any:
+		return "ANY"
+	case One:
+		return "ONE"
+	case Two:
+		return "TWO"
+	case Three:
+		return "THREE"
+	case Quorum:
+		return "QUORUM"
+	case All:
+		return "ALL"
+	case LocalQuorum:
+		return "LOCAL_QUORUM"
+	case EachQuorum:
+		return "EACH_QUORUM"
+	case LocalOne:
+		return "LOCAL_ONE"
+	default:
+		return fmt.Sprintf("invalid consistency: %d", uint16(c))
+	}
+}
+
+func ParseConsistency(s string) (Consistency, error) {
```

Comment:
Parse functions should return err if the provided string is not mapping an enum value

Reply:
yes, `UnmarshalText` is checking if value provided is valid. This case is covered in unit test also

Reply:
ok perfect. I missed that